internal/tfdiags/diagnostics.go

```diff
@@ -101,6 +102,23 @@ func (diags Diagnostics) Append(new ...interface{}) Diagnostics {
 	return diags
 }
 
+func (diags Diagnostics) Merge(other Diagnostics) Diagnostics {
```

Comment:
Perhaps name this StrictDedeuplicate or similar to signal that it is made for a very specific purpose?

Reply:
Changed it to StrictDeduplicateMerge